codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/endpoint/Util.kt

```diff
@@ -53,16 +52,14 @@ class Types(runtimeConfig: RuntimeConfig) {
     val resolveEndpointError = smithyHttpEndpointModule.resolve("ResolveEndpointError")
 }
 
-private fun String.stringToRustName(): String = RustReservedWords.escapeIfNeeded(this.toSnakeCase())
-
```

Comment:
this was private for a reason—we should be quite careful about converting strings to rust identifiers in this way. In general, names should be "symbol aware" so that the symbol provider can be the source of truth.

In the case of test modules, this is still appropriate to use but if you make it public, we should give it a name that resists misuse (and a comment explaining why this generally shouldn't be used to name symbols).